src/lib_json/json_reader.cpp

```diff
@@ -1860,10 +1855,10 @@ String OurReader::getFormattedErrorMessages() const {
   return formattedMessage;
 }
 
-std::vector<OurReader::StructuredError> OurReader::getStructuredErrors() const {
-  std::vector<OurReader::StructuredError> allErrors;
+std::vector<CharReader::StructuredError> OurReader::getStructuredErrors() const {
```

Comment:
The api before this change had an OurReader::getStructuredErrors that returned a std::vector<OurReader::StructuredError>. But now the return type has changed its name, which is an API change, which was I thought what we were trying to avoid in the original proposal by not changing the offsets of the virtual functions in the CharReader class.

Reply:
I've moved those data members and made them private.

I resolved the api change by adding a transformation in OurImpl::getStructuredErrors to convert the CharReader vector to an OurReader vector.

Reply:
@BillyDonahue can you let me know when you get a chance to take a look at this?